src/Development/IDE/Core/Compile.hs

```diff
@@ -447,8 +449,46 @@ getModSummaryFromBuffer fp contents dflags parsed = do
     where
       sourceType = if "-boot" `isSuffixOf` takeExtension fp then HsBootFile else HsSrcFile
 
-
--- | Given a buffer, flags, file path and module summary, produce a
+-- | Given a buffer, env and filepath, produce a module summary by parsing only the imports.
+--   Runs preprocessors as needed.
+getModSummaryFromImports :: FilePath -> Maybe SB.StringBuffer -> HscEnv -> ExceptT [FileDiagnostic] IO ModSummary
```

Comment:
As far as I can see (from quite a brief inspection) the only information ever used from this `HscEnv` is the `DynFlags`, so it should be safe to pass in ` DynFlags` here and then pass in a faked `HscEnv` to `preprocess`.

Reply:
The `HscEnv` is also required for `implicitRequirements`, which is part of the BackPack logic in GHC. Unless BackPack support is a priority, which probably requires a fair amount of work, I think it's safe to drop the call to `implicitRequirements` and generalize the types to lift the requirement of an `HscEnv`